### ISSUE TITLE
Can't sync library

### ISSUE THREAD

**Issue by xkpd3** [haiwen/seafile#1742]
Hi,

I have a library which consists of a folder structure and many 2 to 300MB files. The whole library is 5GB and encrypted. It gets stuck after syncing 112MB.

I can see these lines repeating in the log:
```
[08/27/16 11:46:02] http-tx-mgr.c(1012): Transfer repo '51b0023c': ('normal', 'fs') --> ('normal', 'data')
[08/27/16 11:46:02] http-tx-mgr.c(707): libcurl failed to GET https://seafile.myserver.tld/seafhttp/repo/51b0023c-8abd-4c54-b0a7-1af26d0f77ff/block/8df2067cdb2fdceb16ddbe07313e6bd1a999a414: Transferred a partial file.
[08/27/16 11:46:02] repo-mgr.c(5059): Transfer failed.
[08/27/16 11:46:02] http-tx-mgr.c(707): libcurl failed to GET https://seafile.myserver.tld/seafhttp/repo/51b0023c-8abd-4c54-b0a7-1af26d0f77ff/block/10e050a6620772b17e1b9e8acc56218b3bded9f6: Failed writing received data to disk/application.
[08/27/16 11:46:02] repo-mgr.c(5059): Transfer failed.
[08/27/16 11:46:02] http-tx-mgr.c(707): libcurl failed to GET https://seafile.myserver.tld/seafhttp/repo/51b0023c-8abd-4c54-b0a7-1af26d0f77ff/block/f3880e0bd9cf75d1657f7c69563b86dbb0e7738d: Failed writing received data to disk/application.
[08/27/16 11:46:02] repo-mgr.c(5059): Transfer failed.
[08/27/16 11:46:03] http-tx-mgr.c(707): libcurl failed to GET https://seafile.myserver.tld/seafhttp/repo/51b0023c-8abd-4c54-b0a7-1af26d0f77ff/block/85485292bb0848db511af9f4ad28d965a3916bf3: Failed writing received data to disk/application.
[08/27/16 11:46:03] repo-mgr.c(5059): Transfer failed.
[08/27/16 11:46:03] http-tx-mgr.c(1012): Transfer repo '51b0023c': ('normal', 'data') --> ('error', 'finished')
[08/27/16 11:46:03] clone-mgr.c(819): Transition clone state for 51b0023c from [fetch] to [fetch].
[08/27/16 11:46:03] http-tx-mgr.c(4130): Download with HTTP sync protocol version 1.
[08/27/16 11:46:03] http-tx-mgr.c(1012): Transfer repo '51b0023c': ('normal', 'init') --> ('normal', 'check')
[08/27/16 11:46:03] http-tx-mgr.c(1012): Transfer repo '51b0023c': ('normal', 'check') --> ('normal', 'commit')
[08/27/16 11:46:04] http-tx-mgr.c(1012): Transfer repo '51b0023c': ('normal', 'commit') --> ('normal', 'fs')
```

Do you have any hint how I can debug this and maybe find the file that causes the trouble? I have more libraries (and way bigger ones) that are syncing just fine.

I'm running the 5.1.4 client on OSX and the 5.1.4 64bit server on Debian linux.

[UPDATE]
The library is syncing fine on linux. So I think there is a condition that breaks the OSX client.
[/UPDATE]

Best Regards,
Stefan

**Comment by xkpd3** [comment]
Update:

In the last days, I did the following:
- Upgrade seafile server from 5.1.4 to 6.0.4
- Create a new user + new library
- Copied some files from the old library on OSX and uploaded them into the new library

Now I'm trying to download this new library on Windows. It doesn't work. Even worse - I noticed that the seefile client keeps downloading with max bandwith, but the Library on the disk is not growing.

I see the same log entries I saw earlier on the OSX machine.

```
http-tx-mgr.c(4130): Download with HTTP sync protocol version 1.
[09/26/16 10:09:30] http-tx-mgr.c(1012): Transfer repo 'a7b61cd0': ('normal', 'init') --> ('normal', 'check')
[09/26/16 10:09:31] http-tx-mgr.c(1012): Transfer repo 'a7b61cd0': ('normal', 'check') --> ('normal', 'commit')
[09/26/16 10:09:31] http-tx-mgr.c(1012): Transfer repo 'a7b61cd0': ('normal', 'commit') --> ('normal', 'fs')
[09/26/16 10:09:31] http-tx-mgr.c(1012): Transfer repo 'a7b61cd0': ('normal', 'fs') --> ('normal', 'data')
[09/26/16 10:09:49] http-tx-mgr.c(707): libcurl failed to GET https://cloud.herrhagen.de/seafhttp/repo/a7b61cd0-9840-4775-81a8-e9347fc48533/block/bb9cd3a64b3631cdc4938a41b0d7bf9433f01535: Transferred a partial file.
[09/26/16 10:09:49] repo-mgr.c(5059): Transfer failed.
[09/26/16 10:09:49] http-tx-mgr.c(707): libcurl failed to GET https://cloud.herrhagen.de/seafhttp/repo/a7b61cd0-9840-4775-81a8-e9347fc48533/block/6f973b8576f7e5d86abea4b062f346e7ea04289c: Failed writing received data to disk/application.
[09/26/16 10:09:49] repo-mgr.c(5059): Transfer failed.
[09/26/16 10:09:49] http-tx-mgr.c(707): libcurl failed to GET https://cloud.herrhagen.de/seafhttp/repo/a7b61cd0-9840-4775-81a8-e9347fc48533/block/29feaf2f3b29ea12b595d2d7b013a8f0648c997e: Failed writing received data to disk/application.
[09/26/16 10:09:49] repo-mgr.c(5059): Transfer failed.
[09/26/16 10:09:49] http-tx-mgr.c(707): libcurl failed to GET https://cloud.herrhagen.de/seafhttp/repo/a7b61cd0-9840-4775-81a8-e9347fc48533/block/b726bd404cbaaebce115a30802037e0ac9a5991f: Failed writing received data to disk/application.
[09/26/16 10:09:49] repo-mgr.c(5059): Transfer failed.
[09/26/16 10:09:49] http-tx-mgr.c(1012): Transfer repo 'a7b61cd0': ('normal', 'data') --> ('error', 'finished')
```

I think it has to do with syncing to different file systems.

My wish would be:
- If files can not be synced, skip them
- Show sync issues in the log and in the client in a detail level, that an end-user can resolve it.

Best Regards,
Stefan

**Comment by killing** [comment]
I think it could be a network issue. From libcurl's manual, "transferred a partial file" means:

```
CURLE_PARTIAL_FILE (18) 
   A file transfer was shorter or larger than expected. This happens when the 
   server first reports an expected transfer size, and then delivers data that 
   doesn't match the previously given size.
```

It could be the transfer is somehow interrupted by the server. Can you check your Nginx/Apache log for clues?

**Comment by xkpd3** [comment]
Thank you. I'm using the lighttpd as webserver. I switched on debugging for request and response header, the reverse proxy and rewrite. No clue what's going on and no hint in there about a request that may have gotten cut. The faulty request looks exactly the same as all other requests that are getting accepted from seafile.

I did the effort to setup a new IP and run nginx for seafile there. The issue went away, so it's surely something in the lighttpd proxy module.

I think after having so much trouble (this is not the only issue) I'm going to move away from lighttpd where possible.

The issue can be closed.
